js/clinicalTimeline.js

```diff
@@ -884,24 +884,32 @@ window.clinicalTimeline = (function(){
               tickValues.push(i * maxTime.daysPerYear);
           }
       } else if (zoomLevel === "months") {
-          tickValues.push(parseInt(beginning));
-          for (i=minTime.m + minTime.y * 12 + 1; i < maxTime.m + maxTime.y * 12 - 1; i++) {
+          if (minTime.d !== 0) {
+            beginning -= minTime.d < 0 ? daysPerMonth : -daysPerMonth;
+            minTime = daysToTimeObject(parseInt(beginning));
+            ending += maxTime.d > 0 ? daysPerMonth : -daysPerMonth;
+            maxTime = daysToTimeObject(parseInt(ending));
+          }
```

Comment:
You might want to generalize this problem. What we want is roundDown for start and roundUp for end based on a tick length. Realize that you are also going to need to do the same for 10days and 3days. You might want to create a function similar to http://stackoverflow.com/questions/3407012 so it's more general. I think in our case it makes sense to change the time object in `daysToTimeObject` to have some convenience functions, such as `time.inDays()` and `time.inMonths()` and `time.inYears()`. The for loop below could then be something like:
```js
for (i=roundDown(minTime.inMonths(), 1); i <= roundUp(maxTime.inMonths(), 1); i++) {
  tickValues.push(i * maxTime.daysPerMonth + parseInt(i/12) * 5);
}
```
Obviously in this case you could use `Math.floor` and `Math.ceil`, but for the 3days and 10days cases you would need to create the extra functions `roundUp` and `roundDown`.